src/main/java/biomesoplenty/common/world/BOPBiomeManager.java

```diff
@@ -28,23 +27,18 @@ public static BiomeGenBase createAndRegisterBiome(Class<? extends BiomeGenBase>
 		if (biome != null)
 		{
 			BiomeEntry entry = new BiomeEntry(biome, getConfiguredWeight(biome, biomeType, weight));
-
-			if (BOPConfigurationBiomeGen.config.get(biomeType + " Biomes To Generate", biome.biomeName, true).getBoolean(false))
-			{
-				if (biomeList != null) biomeList.add(entry);
-			}
-
-			return biome;
+			biomeList.add(entry);
 		}
-		
-		return null;
+		return biome;
 	}
 	
 	public static BiomeGenBase createBiome(Class<? extends BiomeGenBase> biomeClass, String biomeName, int biomeId)
 	{
-		if (biomeId == -1) biomeId = BOPConfigurationIDs.config.get("Biome IDs", biomeName + " ID", getNextFreeBiomeId()).getInt();
+		if (biomeId == -2) biomeId = BOPConfigurationIDs.config.get("Biome IDs", biomeName + " ID", getNextFreeBiomeId()).getInt();
+		
+		if (biomeId == -1) return null;
 
-		if (biomeId != -1)
+		if (-1 > biomeId && biomeId < 256)
```

Comment:
(-1 > biomeId && biomeId < 256) is not a meaningful test, since -1 < biomeID is equivalent to biomeID < -1, and thus the <256 becomes redundant.